includes/server/class-llms-rest-memberships-controller.php

```diff
@@ -0,0 +1,590 @@
+			'membership_cat' => 'categories',
+			'membership_tag' => 'tags',
+		);
+
+		return isset( $taxonomy_base_map[ $base ] ) ? $taxonomy_base_map[ $base ] : $base;
+	}
+
+	/**
+	 * Prepares a single post for create or update.
+	 *
+	 * @since [version]
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 * @return array|WP_Error Array of llms post args or WP_Error.
+	 */
+	protected function prepare_item_for_database( $request ) {
+		$prepared_item = parent::prepare_item_for_database( $request );
+		$schema        = $this->get_item_schema();
+
+		// Restriction action.
+		if ( ! empty( $schema['properties']['restriction_action'] ) && isset( $request['restriction_action'] ) ) {
+			$prepared_item['restriction_redirect_type'] = $request['restriction_action'];
+		}
+
+		// Restriction message.
+		if ( ! empty( $schema['properties']['restriction_message'] ) && isset( $request['restriction_message'] ) ) {
+			if ( is_string( $request['restriction_message'] ) ) {
+				$prepared_item['restriction_notice'] = $request['restriction_message'];
+			} elseif ( isset( $request['restriction_message']['raw'] ) ) {
+				$prepared_item['restriction_notice'] = $request['restriction_message']['raw'];
+			}
+		}
+
+		// Restriction page id.
+		if ( ! empty( $schema['properties']['restriction_page_id'] ) && isset( $request['restriction_page_id'] ) ) {
+			$page = get_post( $request['restriction_page_id'] );
+			if ( $page && is_a( $page, 'WP_Post' ) ) {
+				$prepared_item['redirect_page_id'] = $request['restriction_page_id'];
+			} else {
+				$prepared_item['redirect_page_id'] = 0;
+			}
+		}
+
+		// Restriction URL.
+		if ( ! empty( $schema['properties']['restriction_url'] ) && isset( $request['restriction_url'] ) ) {
+			$prepared_item['redirect_custom_url'] = $request['restriction_url'];
+		}
+
+		// Sales page id.
+		if ( ! empty( $schema['properties']['sales_page_page_id'] ) && isset( $request['sales_page_page_id'] ) ) {
+			$page = get_post( $request['sales_page_page_id'] );
+			if ( $page && is_a( $page, 'WP_Post' ) ) {
+				$prepared_item['sales_page_content_page_id'] = $request['sales_page_page_id'];
+			} else {
+				$prepared_item['sales_page_content_page_id'] = 0;
+			}
+		}
+
+		// Sales page type.
+		if ( ! empty( $schema['properties']['sales_page_type'] ) && isset( $request['sales_page_type'] ) ) {
+			$prepared_item['sales_page_content_type'] = $request['sales_page_type'];
+		}
+
+		// Sales page URL.
+		if ( ! empty( $schema['properties']['sales_page_url'] ) && isset( $request['sales_page_url'] ) ) {
+			$prepared_item['sales_page_content_url'] = $request['sales_page_url'];
+		}
+
+		return $prepared_item;
+
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param LLMS_Membership $membership LLMS Membership.
+	 * @return array Links for the given object.
+	 */
+	protected function prepare_links( $membership ) {
+		$parent_links = parent::prepare_links( $membership );
+		$id           = $membership->get( 'id' );
+
```

Comment:
Memberships have no content link, see: https://github.com/eri-trabiccolo/lifterlms-rest/blob/memberships/includes/server/class-llms-rest-memberships-controller.php#L590

Reply:
Thanks for catching the unneeded content link.

It looks like I left out `access_plans` and `auto_enrollment_courses` links. I don't see `collection`, and `self` links in your code. Should I add those as well?

Reply:
I found `collection` and `self` in `LLMS_REST_Controller->prepare_links()`.